spec/features/work_packages/tabs/relations_children_spec.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+require "support/edit_fields/edit_field"
+
+RSpec.describe "Relations children tab", :js, :with_cuprite do
+  shared_let(:normal_cf) { create(:string_wp_custom_field, is_required: false) }
+  shared_let(:required_cf) { create(:string_wp_custom_field, is_required: true) }
+  shared_let(:type_milestone) { create(:type_milestone) }
+  shared_let(:type_task) { create(:type, name: "Task", custom_fields: [normal_cf]) }
+  shared_let(:type_risk) { create(:type, name: "Risk", custom_fields: [required_cf]) }
+  shared_let(:default_status) { create(:default_status) }
+  shared_let(:default_priority) { create(:default_priority) }
+
+  shared_let(:project) do
+    create(:project,
+           types: [type_task, type_risk, type_milestone],
+           work_package_custom_fields: [normal_cf, required_cf])
+  end
+
+  shared_let(:work_package) { create(:work_package, type: type_task, project:, subject: "Parent") }
+
+  let(:relations_tab) { Components::WorkPackages::Relations.new(work_package) }
+  let(:create_dialog) { Components::WorkPackages::CreateDialog.new }
+  let(:wp_page) { Pages::FullWorkPackage.new(work_package, project) }
+
+  current_user { user }
+
+  context "with permissions to add children" do
+    let!(:user) do
+      create(:user, member_with_permissions: { project => %i[view_work_packages add_work_packages manage_subtasks] })
+    end
+
+    it "can add a new child" do
+      wp_page.visit_tab!("relations")
+      relations_tab.expect_add_relation_button
+      relations_tab.expect_new_relation_type("New child")
+      relations_tab.expect_new_relation_type("Existing child")
+
+      relations_tab.select_relation_type "New child"
+
+      create_dialog.select_type "Task"
+      create_dialog.set_subject "Hello there"
+      create_dialog.set_description "Some *markdown* content"
+      create_dialog.expect_no_custom_field(normal_cf)
+
+      create_dialog.select_type "Risk"
+      # Retains subject and description
+      create_dialog.expect_subject "Hello there"
+      create_dialog.expect_description "Some markdown content"
+
+      # Shows a custom field
+      create_dialog.set_custom_field(required_cf, "Custom value")
+      create_dialog.submit
+
+      wait_for_network_idle
+
+      page.within("#work-package-relations-tab-content") do
+        expect(page).to have_content("Hello there")
+        expect(page).to have_content("RISK")
+      end
+    end
+
+    context "when work package is a milestone" do
+      let(:work_package) { create(:work_package, type: type_milestone, project:, subject: "Parent") }
+
+      it "does not show the action" do
+        wp_page.visit_tab!("relations")
+        relations_tab.expect_no_add_relation_button
+      end
```

Comment:
I did not understand that test at first, because the "+ Relation" button should be there to add other kind of relations, but actually it's also missing the `:manage_work_package_relations` permission, which is why it's not there.

Not sure how that can be improved. A comment?